pom.xml

```diff
@@ -153,7 +152,42 @@
 			</plugin>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-compiler-plugin</artifactId>
+				<artifactId>maven-checkstyle-plugin</artifactId>
+				<version>${checkstyle.version}</version>
+				<executions>
+					<execution>
+						<id>verify-style</id>
+						<phase>process-classes</phase>
+						<goals>
+							<goal>check</goal>
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<logViolationsToConsole>true</logViolationsToConsole>
+					<checkstyleRules>
+						<module name="Checker">
+							<!-- Checks for Size Violations.                    -->
+							<!-- See http://checkstyle.sf.net/config_sizes.html -->
+							<module name="FileLength">
+								<property name="max" value="500" />
+								<property name="fileExtensions" value="java"/>
+							</module>
+							<!-- Checks for long methods and constructors.                           -->
+							<!-- See https://checkstyle.sourceforge.io/config_sizes.html#MethodLength -->
+							<module name="TreeWalker">
+								<module name="MethodLength">
+									<property name="tokens" value="METHOD_DEF"/>
+									<property name="max" value="35"/>
```

Comment:
...and 20 for this one?

Reply:
Same as FileLength we have methods with a maximum length of 31 hence I have entered 35.

Reply:
Can you point out a method with 31 lines?

Reply:
```
[ERROR] target/generated-sources/annotations/com/ideal/studentlog/helpers/mappers/TeacherMapperImpl.java:[15,5] (sizes) MethodLength: Method length is 31 lines (max allowed is 20).
[ERROR] target/generated-sources/annotations/com/ideal/studentlog/helpers/mappers/StudentMapperImpl.java:[15,5] (sizes) MethodLength: Method length is 27 lines (max allowed is 20).
[ERROR] target/generated-sources/annotations/com/ideal/studentlog/helpers/mappers/StudentApplicationMapperImpl.java:[16,5] (sizes) MethodLength: Method length is 33 lines (max allowed is 20).
[ERROR] target/generated-sources/annotations/com/ideal/studentlog/helpers/mappers/AdminMapperImpl.java:[15,5] (sizes) MethodLength: Method length is 29 lines (max allowed is 20).
```

Reply:
Yikes... can we omit generated and test sources?

Reply:
https://stackoverflow.com/questions/18013322/checkstyle-exclude-folder

Reply:
Looking into it. Thanks for the related link, otherwise it would have been hard for me to understand.